### PR TITLE
enhance cinn compilation as third_party of paddle

### PR DESCRIPTION
**Pull request by zhiqiu**
- set project language to detect cxx and cc
- be able to accept specified PYTHON_BINARY envs

### CODE REVIEW

**Comment by zhhsplendid** [comment]
close and reopen, try for trigger CI

File: CMakeLists.txt [hunk 1]
```diff
@@ -112,8 +112,12 @@ add_subdirectory(tutorials)
 set(core_src "${cinnapi_src}")
 
 set(Python_VIRTUALENV FIRST)
+if(NOT PYTHON_EXECUTABLE)
 find_package(PythonInterp ${PY_VERSION} REQUIRED)
```
Comment (zhhsplendid) [inline CMakeLists.txt:116]: Add space at the beginning of line 116 and line 119?

**Comment by zhhsplendid** [comment]
close and reopen to trigger CI

File: CMakeLists.txt (hunk 1 above)
Comment (zhiqiu) [reply CMakeLists.txt:116]: I will refine it in the next pr.

**Review by thisjiang — APPROVED** [review]
LGTM

**Pull request merged**